File: Daily Challenges/2020/August/Non-overlapping Intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        n = len(intervals)
        intervals.sort(key = lambda x : (x[1], x[0]))
        
        if n == 0:
            return 0
        
        ans = [intervals[0]]
        for i in range(1, n):
            if intervals[i][0] >= ans[-1][1]:
                ans.append(intervals[i])
        
        return n - len(ans)

File: daily_challenge_during_phd/2023/July/Non-overlapping Intervals.cpp
bool cmp(vector<int> &a, vector<int> &b) {
    return a[1] < b[1];
}

class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        if(intervals.size() == 0) return 0;

        sort(intervals.begin(), intervals.end(), cmp);

        int taken = 0, last_intervals_end_time = -60000;
        for(int i = 0; i < intervals.size(); i++) {
            // cout << "i " << i << " " << last_intervals_end_time << endl;
            if(intervals[i][0] >= last_intervals_end_time)
                taken++, last_intervals_end_time = intervals[i][1]; 
        }

        return intervals.size() - taken;
    }
};
